src/fcio.c

```diff
@@ -356,7 +356,15 @@ char asciiToPetscii(byte c)
     if (c >= 192) {
         return c - 128;
     }
-    return c;
+    return (char)c;
+}
+
+#ifdef __llvm__
```

Comment:
`__clang__` might be preferred over `__llvm__` as it will then also be recognised by the Calypsi compiler.

Reply:
Also, according to the standard it's better to add the deprecation attribution to the function declaration, i.e. `fcio.h`:
_"Indicates that the name or entity declared with this attribute is [deprecated](https://en.wikipedia.org/wiki/Deprecation)"_

Reply:
Good points, while doing that I noticed I was deprecating the wrong function -- this one was never even exposed through the header file. Made some corrections now, taking into account your comments.